src/Core/Checkout/Customer/SalesChannel/RegisterRoute.php

```diff
@@ -339,7 +341,15 @@ private function validateRegistrationData(
         }
 
         $accountType = $data->get('accountType', CustomerEntity::ACCOUNT_TYPE_PRIVATE);
-        if ($billingAddress instanceof DataBag || !($shippingAddress instanceof DataBag)) {
+
+        // The billing address is mandatory.
+        // The shipping address is optional but if there is one, a non-array value results in an exception in the validation process.
+        $definition->add('billingAddress', new Type('associative_array'));
+        $definition->add('shippingAddress', new AtLeastOneOf([new Type('associative_array'), new IsNull()]));
+
+        // The billing address validation must not be added if the data is neither a data bag nor valid because the validation building will fail.
+        // Using a null value must be possible to allow the event based modification (see BuildValidationEvent).
+        if ($billingAddress instanceof DataBag || (!$shippingAddress instanceof DataBag && $billingAddress === null)) {
```

Comment:
If shippingAddress is NOT an instance of DataBag it is NULL?

So would that condition be more readable?
```suggestion
        if ($billingAddress instanceof DataBag || ($shippingAddress === null && $billingAddress === null)) {
```

Reply:
The shipping address can be a string as well. I **think** the billing address should be "used" (validation enforced) if the shipping address is invalid (I am not sure about the intention here). And its also aligned with the next check for `$shippingAddress instanceof DataBag`.